#!/usr/bin/env python
PACKAGE = "decision"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

## add gen.add() smparm=gen.enum() 
#       Name                    Type      Reconfiguration level  
#       Description  
#       Default  Min  Max  
gen.add("params_modify_switch", bool_t,   0,
        "params_modify_switch",
        False)
gen.add("control_modify_switch", bool_t,   0,
        "control_modify_switch",
        False)
gen.add("previewInner_param",double_t,0,
        "previewInner_param",
        0.5,0.5,2)
gen.add("previewCurvature0_param",double_t,0,
        "previewCurvature0_param",
        13,10,15)
gen.add("previewCurvature5_param",double_t,0,
        "previewCurvature5_param",
        0,0,2)
gen.add("pathNumber",int_t,0,
        "pathNumber",
        1,1,2)
gen.add("targetSpeed",double_t,0,
        "targetSpeed",
        0,0,40)
gen.add("targetSteeringAngle",double_t,0,
        "targetSteeringAngle",
        0,-540,540)
gen.add("targetSteeringVelocity",double_t,0,
        "targetSteeringVelocity",
        10,100,360)

gen.add("kp_large",double_t,0,
        "kp_large",
        500,100,1500)
gen.add("kd_large",double_t,0,
        "kd_large",
        500,0,800)
gen.add("kp_small",double_t,0,
        "kp_small",
        300,100,1500)
gen.add("kd_small",double_t,0,
        "kd_small",
        300,0,800)
gen.add("ki_test",double_t,0,
        "ki_test",
        0,0,300)
gen.add("FILTER_RATE",double_t,0,
        "FILTER_RATE",
        0.8,0,1)
gen.add("extend_length",double_t,0,
        "extend_length",
        6,0,20)
gen.add("path_record_switch", bool_t,   0,
        "path_record_switch",
        False)

exit(gen.generate(PACKAGE, "decision", "decision_param"))